shmem.c

```diff
@@ -155,12 +155,16 @@ pthread_mutex_t create_mutex() {
 	return lock;
 }
 
-void lock_shm() {
+void _lock_shm(const char* function, const int line, const char * file) {
```

Comment:
Provided by a macro in `shmem.h`. Note the `_` at the beginning of the name to symbolize this.